### ISSUE TITLE
Pairwise losses are calculated incorrectly.

### ISSUE THREAD

**Issue by HarrieO** [tensorflow/ranking#73]
The pairwise losses, i.e. 'pairwise_hinge_loss' and 'pairwise_logistic_loss', should not be affected by the values of labels beyond their ordering.
Since these loss functions are based on the indicator function I(yn < ym) only the ordering of labels matters.
However, multiplying your labels by a constant factor will result in a different loss.
Similarly, other changes that maintain ordering affect the loss as well.

(Practical note: Scaling by a constant factor results in the same gradient but scaled, other transformations, i.e. taking the label squared, can lead to substantially different gradients.)

As far as I can tell this error is caused in this function: https://github.com/tensorflow/ranking/blob/8536405845e8ca180dae01c4d0e34ecb1be89b65/tensorflow_ranking/python/losses.py#L551
where a pairwise loss is always weighted by the absolute difference of label values in line:
https://github.com/tensorflow/ranking/blob/8536405845e8ca180dae01c4d0e34ecb1be89b65/tensorflow_ranking/python/losses.py#L588
Instead it should be weighted by binary label difference indicators.

To reproduce this error, take any set of queries, calculate the loss, then change your data by multiplying the labels by 100, and recalculate the loss.
You will find the loss is now 100x as large.

**Comment by xuanhuiwang** [comment]
HarrieO@, your observation is correct. We take the label difference into account. This shouldn't affect anything for binary labels, but it indeed has the scaling effect and can give different results for label squares.

I agree that only binary indicator is needed for ordering effect. The LambdaLoss paper shows that the current implementation is a bound for ARP: https://storage.googleapis.com/pub-tools-public-publication-data/pdf/1e34e05e5e4bf2d12f41eb9ff29ac3da9fdb4de3.pdf.

However, I think it is better to use the binary indicator to be aligned with literature. We will modify our losses accordingly.

**Comment by we1559** [comment]
@xuanhuiwang
It seems that we only need to remove this code.
https://github.com/tensorflow/ranking/blob/8536405845e8ca180dae01c4d0e34ecb1be89b65/tensorflow_ranking/python/losses.py#L545

In my opinion, it only affacts the pure pairwise loss without lambda weight. Pairwise loss with lambda weight is calculated correct.

I'd like to submit a pull request if my solution is right.

**Comment by xuanhuiwang** [comment]
@we1559, the proposal looks right.

There additional work to adapt the test cases after this change. Feel free to give it a try and send the PR over.

**Comment by we1559** [comment]
@xuanhuiwang 
I've sent the PR. https://github.com/tensorflow/ranking/pull/75

**Comment by xuanhuiwang** [comment]
PR is merged and fixed.

**Comment by MiladShahidi** [comment]
I am using pairwise logistic loss and trying to overfit a batch of two examples and reach the lowest possible loss (following the advice [here](http://karpathy.github.io/2019/04/25/recipe/)).

Setting the implementation aside for a moment, and looking at equation for pairwise logistic loss (eq. 9) in the TF-Ranking paper, I'm wondering what the lowest achievable loss is in this case.

Here is why I'm having a difficult time with this: As long as the ordering predicted by logits is correct, you can make the loss arbitrarily small or large by scaling the logits. Even when the y_hat vector is identical to y, the loss is not necessarily zero. So, wouldn't make sense to say the lowest possible loss is zero.

I'm using MovieLens where the labels are integers in the [1, 5] range and I have a Sigmoid activation in the last layer (group_size=1) which results in y_hats that are between 0 and 1. and I noticed that I cannot drive the loss down to zero on a small batch. However, when I change the last activation to linear the loss does go down to zero because y_hat is free to scale as much as needed to make the loss arbitrarily small. But this scaling will not result in improved ranking.
